src/main/java/com/netflix/imflibrary/st2067_2/Application2E2021.java

```diff
@@ -514,12 +516,26 @@ private static boolean validateHT(CompositionImageEssenceDescriptorModel imageDe
         }
 
         int codestreamB = (p.cap.ccap[0] & 0b11111) + 8;
-        if (codestreamB > maxB) {
+
+        /*
+        *   NOTE: The Parameter B constraints in ST 2067-21:2023 are arguably too narrow, and existing implementations do violate them under certain circumstances.
+        *   Since practical issues are not expected from software decoders for Parameter B values <= 24, we only raise WARNING for spec violations below that threshold.
+        *
+        *   TODO: This should be revisited as more implementations become available.
+        */
+
+        if (codestreamB > 24) {
+            logger.addError(
+                    IMFErrorLogger.IMFErrors.ErrorCodes.APPLICATION_COMPOSITION_ERROR,
+                    IMFErrorLogger.IMFErrors.ErrorLevels.FATAL,
+                    "APP2.HT: Parameter B has exceeded its limit to an extend that decoder issues are to be expected");
+            isValid = false;
+        } else if (codestreamB > maxB) {
             logger.addError(
                 IMFErrorLogger.IMFErrors.ErrorCodes.APPLICATION_COMPOSITION_ERROR,
-                IMFErrorLogger.IMFErrors.ErrorLevels.FATAL,
+                IMFErrorLogger.IMFErrors.ErrorLevels.WARNING,
                 "APP2.HT: Parameter B has exceeded its limits");
-            isValid = false;
+            isValid = true;
```

Comment:
Remove since isValid is true by default and you would not want to overwrite a false value.